### PR TITLE
Move Tour to shepherd.js

### PR DESCRIPTION
**Pull request by mimi1vx**
Bootstrap-Tour isn't supported anymore and blocks bootstrap upgrade

See: https://progress.opensuse.org/issues/101385

### CODE REVIEW

**Comment by Martchus** [comment]
Thanks for this and unblocking the bootstrap upgrade.

The styling looks a bit out of place as everything is a bit bigger than normal page elements:
screenshot_20220303_171040

The missing space between the checkbox and its label also doesn't look pretty.

Otherwise it seems to be functioning as expected except that it doesn't store the value where it left off correctly (and also not `0` for dismissing it permanently; but I suppose that's why it is still WIP).

**Comment by mimi1vx** [comment]
> Thanks for this and unblocking the bootstrap upgrade.
> 
> The styling looks a bit out of place as everything is a bit bigger than normal page elements: screenshot_20220303_171040
> 
> The missing space between the checkbox and its label also doesn't look pretty.
 .. style is standard shepherd.css now, needs some work 

> Otherwise it seems to be functioning as expected except that it doesn't store the value where it left off correctly (and also not `0` for dismissing it permanently, but I suppose that's why it is still WIP).

any idea why `postFeature` always sends `latestFeature` ?

File: assets/javascripts/feature.js
Comment (Martchus) [inline assets/javascripts/feature.js:5]: Just for the record: As mentioned in chat, `$('#dont-notify').is(':checked')` does not work because at the time this function runs `$('#dont-notify')` is not part of the DOM anymore.

**Review by Martchus — CHANGES_REQUESTED** [review]

File: assets/javascripts/feature.js
Comment (Martchus) [inline assets/javascripts/feature.js:16]:
```suggestion
  checkCheckbox = checkboxElement.checked;
```

Although I would just inline it completely, see the other comment.

File: assets/javascripts/feature.js
Comment (Martchus) [inline assets/javascripts/feature.js:61]: I'd say this can be inlined and the variable should have a better name:

```suggestion
      text: "<p>Access the job group overview pages from here. Besides test results, a description and commenting area are provided.</p><div class='checkbox'><label><input type='checkbox' id='dont-notify' onchange='window.dismissPermanently = this.checked'> Don't notify me anymore (permanent)</label></div>",
```

**Review by Martchus — CHANGES_REQUESTED** [review]

**Comment by mimi1vx** [comment]
New look:

Snímek obrazovky 2022-03-09 v 15 27 40

**Review by Martchus — APPROVED** [review]

**Review by okurz — CHANGES_REQUESTED** [review]
Change looks ok. Some tests are failing and the PR is still marked WIP hence not approving

**Comment by mimi1vx** [comment]
Is marked WIP because tests still missing

**Comment by Martchus** [comment]
Note that the package build is bound to fail because the cache needs to be updated (https://open.qa/docs/#update-asset-cache). So we need to force merge this if everything else passes and then update the rpm cache.

**Comment by mimi1vx** [comment]
@okurz  @Martchus  `t/ui/21-admin-needles.t` fail isn't related to this change, I got the same fail on master. 

only need check fullstack test

**Comment by mimi1vx** [comment]
> Just for the record, that was the failure:
> 
> ```
> [15:08:26] t/ui/21-admin-needles.t .................... 19/? clickElement: element click intercepted: Element <button type="button" id="btn_custom_last_seen" class="btn" title="Apply">...</button> is not clickable at point (298, 252). Other element would receive the click: <footer class="shepherd-footer">...</footer> at /home/squamata/project/t/ui/../lib/OpenQA/SeleniumTest.pm:81 at /home/squamata/project/t/ui/../lib/OpenQA/SeleniumTest.pm line 84.
> 	OpenQA::SeleniumTest::__ANON__(Test::Selenium::Chrome=HASH(0x55d63b3059c8), "Error while executing command: clickElement: element click in"..., HASH(0x55d63b5546e8)) called at /usr/lib/perl5/vendor_perl/5.26.1/Selenium/Remote/Driver.pm line 356
> 	Selenium::Remote::Driver::catch {...} ("Error while executing command: clickElement: element click in"...) called at /usr/lib/perl5/vendor_perl/5.26.1/Try/Tiny.pm line 123
> 	Try::Tiny::try(CODE(0x55d63c931288), Try::Tiny::Catch=REF(0x55d63c9273f8)) called at /usr/lib/perl5/vendor_perl/5.26.1/Selenium/Remote/Driver.pm line 361
> 	Selenium::Remote::Driver::__ANON__(CODE(0x55d63b030fc0), Test::Selenium::Chrome=HASH(0x55d63b3059c8), HASH(0x55d63b5546e8)) called at (eval 1699)[/usr/lib/perl5/vendor_perl/5.26.1/Class/Method/Modifiers.pm:89] line 1
> 	Selenium::Remote::Driver::__ANON__(Test::Selenium::Chrome=HASH(0x55d63b3059c8), HASH(0x55d63b5546e8)) called at (eval 1701)[/usr/lib/perl5/vendor_perl/5.26.1/Class/Method/Modifiers.pm:148] line 2
> 	Selenium::Remote::Driver::_execute_command(Test::Selenium::Chrome=HASH(0x55d63b3059c8), HASH(0x55d63b5546e8)) called at (eval 1660)[/usr/lib/perl5/vendor_perl/5.26.1/Sub/Quote.pm:3] line 17
> 	Selenium::Remote::WebElement::_execute_command(Test::Selenium::Remote::WebElement=HASH(0x55d63c932690), HASH(0x55d63b5546e8)) called at /usr/lib/perl5/vendor_perl/5.26.1/Selenium/Remote/WebElement.pm line 63
> 	Selenium::Remote::WebElement::click(Test::Selenium::Remote::WebElement=HASH(0x55d63c932690)) called at t/ui/21-admin-needles.t line 312
> 	main::__ANON__() called at /usr/lib/perl5/5.26.1/Test/Builder.pm line 309
> 	eval {...} called at /usr/lib/perl5/5.26.1/Test/Builder.pm line 309
> 	Test::Builder::subtest(Test::Builder=HASH(0x55d630032c98), "custom needles search", CODE(0x55d63c9083d8)) called at /usr/lib/perl5/5.26.1/Test/More.pm line 807
> 	Test::More::subtest("custom needles search", CODE(0x55d63c9083d8)) called at t/ui/21-admin-needles.t line 366
> ```
> 
> Maybe there's already a ticket about it.

none, what is weird locally it fails with different error ( but same in master and tour branch)

**Comment by okurz** [comment]
OBS checks fail because the new js module isn't accessible there, please see https://github.com/os-autoinst/openQA/pull/4543#issuecomment-1058189097 about updating the asset cache

**Comment by mimi1vx** [comment]
@Martchus @okurz  all test's passed, it looks like something in my test env is broken for 21-admin-needles

**Comment by okurz** [comment]
maybe but we should not merge this PR until all checks pass

**Comment by mimi1vx** [comment]
@okurz checks can't pass on this PR because we need refresh assets cache in package .. which we can't without merge of this PR

File: assets/javascripts/feature.js
Comment (okurz) [inline assets/javascripts/feature.js:1]:
```suggestion
var disablePermanently = false;
```

**Review by okurz — COMMENTED** [review]
Yes, now all other tests are passed so that part is ok. Found one typo in variable name to fix, rest is fine

File: assets/javascripts/feature.js
Comment (Martchus) [reply assets/javascripts/feature.js]: It doesn't look resolved, I still see `Pernamently`.

Comment (mimi1vx) [reply assets/javascripts/feature.js]: i hate words with *manent* ...

**Pull request merged**
